### ISSUE TITLE
[Request] Share fine-tuning training scripts

### ISSUE THREAD

**Issue by lileetung** [fahmiaziz98/receipt_parsing#1]
Hello! I’ve been exploring Donut OCR for receipt parsing and was very impressed by the fine-tuning results demonstrated in your repository.

To better understand and reproduce your workflow, could you please share:

1. The complete training script or notebook  
2. Configuration details (learning rate, batch size, number of epochs, etc.)  
3. Environment requirements (transformers version, PyTorch version, etc.)

So far, I have:

- Downloaded the official Donut base model  
- Prepared and labeled a dataset of ~500 receipt images  
- Run the baseline training on a Colab A100 instance  

Despite these efforts, the average edit distance on my validation set has not improved and remains around 0.48.

I would greatly appreciate any guidance or suggestions you can provide!  
Thank you for open-sourcing this fantastic project. 😊

You can reach me at: leetung0718@gmail.com

**Comment by fahmiaziz98** [comment]
Sorry, I haven't updated the repository for a long time so I don't know if this notebook has errors,

So this is my notebook, actually this notebook is from the official donut and I just tried to change some parameters, everything is in the notebook, since you have an A100 try increasing the epoch, changing the lr etc.

https://colab.research.google.com/drive/1YPAutuI25By25F5DF4srOoy0l1gwmA_k?usp=sharing
official donut: https://github.com/clovaai/donut?tab=readme-ov-file

**Comment by lileetung** [comment]
Hi there,

Thank you so much for sharing your notebook and parameter settings! 😊 I followed your Colab link and the official Donut example, copied everything into my Colab A100 environment. Here’s what I observed:

Your run: After the very first training iteration (Epoch 0), you achieved a remarkably low average edit distance on the validation set (ED < 0.02).
<img width="930" alt="Image" src="https://github.com/user-attachments/assets/5c8bf2bc-be26-4c09-aad3-db9532a802e0" />

My run: Using the exact same code flow, but my ED is 0.69.

<img width="841" alt="Image" src="https://github.com/user-attachments/assets/aba148a0-2b22-4776-9219-2101968ee097" />

Did you perform any special processing on the Donut model before training? For example, did you freeze certain layers, run additional pre-training, or modify the parameter initializations?

Thanks again for open-sourcing this fantastic project and for your time in answering these questions! I’m excited to get closer to your fast convergence. 🙏

**Comment by fahmiaziz98** [comment]
I think there was a mistake that time, I forgot because it's been a long time.

how about increasing epoch (e.g. 30)? changing lr or Optimizer? is there any decrease in ED?

try changing this part
```py
config = VisionEncoderDecoderConfig.from_pretrained("naver-clova-ix/donut-base")
# Update image size
config.encoder.image_size = image_size # (height, width)
# update max_length of the decoder (for generation)
config.decoder.max_length = max_length # <-- using decoder not encoder
```
and in the image Preprocessing section
```py
processor.feature_extractor.size = image_size[::-1]
processor.feature_extractor.do_align_long_axis = False
processor.feature_extractor.do_normalize = True
processor.feature_extractor.     # <--- you can add/replace it
```
